mlflow/sklearn/__init__.py

```diff
@@ -518,6 +518,12 @@ def __init__(self, sklearn_model):
             if fn := getattr(self.sklearn_model, predict_fn, None):
                 setattr(self, predict_fn, fn)
 
+    def get_raw_model(self):
+        """
+        Returns the underlying scikit-learn model.
+        """
+        return self.sklearn_model
```

Comment:
similarly, we can add a `_wrapped_predict_fn` function into the returned raw model object.

Reply:
I don't think we should modify the returned object here, which is unnatural side effect for the function named `get_raw_model`. I agree that the evaluator logic is super long and it can be structured better, but it should be in the evaluator namespace imo (or sth indicates it is used by evaluator) as it is only used for evaluation.

Reply:
I feel like we shouldn't modify the original model by adding a function on the raw xgboost model, while I moved the function into mlflow/xgboost/__init__.py file so we reuse it in the wrapper.